core/listener.go

```diff
@@ -163,6 +171,12 @@ func (cl *Listener) listen(ctx context.Context, sub <-chan types.EventDataSigned
 				return errors.New("underlying subscription was closed")
 			}
 
+			if cl.chainID != "" && b.Header.ChainID != cl.chainID {
```

Comment:
optionally, we can enforce the chainID on the listener. I'm not too picky with this component's code bc it will be removed hopefully soon ❤️